### ISSUE TITLE
Postgres schema support

### ISSUE THREAD

**Issue by patrick-jones** [TeamHG-Memex/agnostic#1]
I am building a multi-tenant application where each tenant's data will be stored in their own postgres schema. Note that in postgres, a schema != database, it is more like a namespace within a database. See, for example: https://github.com/radiant/radiant/wiki/Using-multiple-PostgreSQL-schemas

Anyway, I would like to use a single admin user to be able to run the same set of migrations repeatedly targeting all of the different schemas. To do this, we have to be able to change the database connection's [search path](http://www.postgresql.org/docs/current/static/ddl-schemas.html) before running the migration scripts.

Looking at the code, it looks like this would come down to:

* Adding a configuration variable for schema (unfortunately, "schema" is already used for "database")
* Add code at connection time to issue the proper "SET search_path..." commands.

An alternative would be be to allow some kind of pre-migration code hook where you could do whatever you wanted to the connection.

**Comment by mehaase** [comment]
I like your suggestion: the database name option should be called `--database` and `--schema` should mean the schema name. One question: should `--schema` take a list of schema names, like `--schema foo,bar,public`?

This will require some rewriting of docs and unit tests. Is this something you can send a pull request for? If not, I can get to it sometime in December.

**Comment by patrick-jones** [comment]
I can probably do a pull request, though I might not get to it immediately. I think taking multiple schema names would be useful, as would some kind of ability to run against all non-public schemas.

**Comment by mehaase** [comment]
Ok. Let me know if you do get started on it, so that we don't duplicate effort. If I get started first, I'll let you know.

**Comment by mehaase** [comment]
Hey @patrick-jones I've just pushed an implementation of what we discussed about renaming `--schema` to `--database` and adding a new `--schema` flag in its place. It supports Postgres search paths, including the special `$user` path, e.g. ``--schema '"$user",public'`. I did not implement the "run against all non-public schemas" feature.

I tested this carefully to verify that it works as I think it should, but I don't use schemas myself (I put everything in public), so I could use your feedback on whether this works as *you* expect it to.
